### PR TITLE
Adding negative test cases to deleted from replicas

### PR DESCRIPTION
**Pull request by Bento007**
### Test plan
- Negative test cases have been added to verify that we are unable to delete from the main replica.

### CODE REVIEW

File: tests/test_gshcablobstore.py [hunk 1]
```diff
@@ -17,6 +19,7 @@
 
 class TestGSHCABlobStore(unittest.TestCase, HCABlobStoreTests):
     def setUp(self):
```
Comment (xbrianh) [inline tests/test_gshcablobstore.py]: I'm a little concerned that these tests could become flaky. Let's upload `Deletion_test.txt` here to make sure it always exists.

File: infra/gcp_service_account/main.tf [hunk 2]
```diff
@@ -20,12 +20,6 @@ resource "google_project_iam_member" "cloudruntimeconfiguratoradmin" {
   member  = "serviceAccount:${google_service_account.dss.email}"
 }
 
-resource "google_project_iam_member" "storageadmin" {
-  project = "${data.google_project.project.project_id}"
-  role    = "roles/storage.admin"
-  member  = "serviceAccount:${google_service_account.dss.email}"
-}
-
```
Comment (xbrianh) [inline infra/gcp_service_account/main.tf]: Consider separating `resource "google_project_iam_member" "storageadmin"` removal into it's own PR to make it prominent in the repo history.

**Comment by Bento007** [comment]
the credential used to run the tests have deletion permissions. We need to create a service account that simulated the production service credentials permissions and use those to run the deletion test.

**Comment by xbrianh** [comment]
What's the status of this PR?

**Comment by Bento007** [comment]
unchanged

**Pull request closed without merging**